KEP-4939:Documentation updates for alpha release (#56322)
kubernetes/website: content/en/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes.md

- The probes run against the pod IP address or its hostname.
  Be sure to configure your gRPC endpoint to listen on the Pod's IP address.
- The probes connect over plaintext by default. See
  [Use TLS with gRPC probes](#grpc-probe-tls) for how to enable TLS.
- There are no error codes for built-in probes. All errors are considered as probe failures.
- If `ExecProbeTimeout` feature gate is set to `false`, grpc-health-probe does **not**
  respect the `timeoutSeconds` setting (which defaults to 1s), while built-in probe would fail on timeout.

### Use TLS with gRPC probes {#grpc-probe-tls}

> {{< feature-state for_k8s_version="v1.37" state="alpha" >}}

Reviewer: Use the `feature_gate_name` form so the version and state auto-update when the feature graduates. The concept page already uses this form.

```suggestion
{{< feature-state feature_gate_name="GRPCContainerProbeTLS" >}}
```

Author: done.

> By default the kubelet connects to gRPC health endpoints over plaintext. If

Reviewer: nit: `kubelet` should be in backticks (component binary). 

```suggestion
By default the `kubelet` connects to gRPC health endpoints over plaintext. If
```

Author: done.

your application serves gRPC only over TLS, you can add the `mode` field to the
`grpc` field in the probe specification and specify a value of `TLS`. This
requires the `GRPCContainerProbeTLS`
[feature gate](/docs/reference/command-line-tools-reference/feature-gates/) to
be enabled on both the `kube-apiserver` and the `kubelet`.

> ```yaml

Reviewer: Every other probe example on this page uses `{{% code_sample %}}` with a file under `content/en/examples/pods/probe/` (see `exec-liveness.yaml`, `http-liveness.yaml`, `tcp-liveness-readiness.yaml`, `grpc-liveness.yaml`). This manifest should follow the same pattern — extract it to something like `content/en/examples/pods/probe/grpc-tls-liveness.yaml` and reference it with:

```suggestion
{{% code_sample file="pods/probe/grpc-tls-liveness.yaml" %}}
```

Author: added, thanks

apiVersion: v1
kind: Pod
metadata:
  name: grpc-tls-app
spec:
  containers:
  - name: app

>     image: my-grpc-app:latest

Reviewer: `my-grpc-app:latest` is a placeholder image that does not exist — a reader cannot `kubectl apply` this and see the feature work. Every other probe example on this page uses a real image (`agnhost`, `etcd`).

Consider using `agnhost` with `grpc-health-checking`, which provides a gRPC health endpoint. For example:

```suggestion
    image: registry.k8s.io/e2e-test-images/agnhost:2.40
```

With appropriate `args` for the grpc-health-checking subcommand. See `http-liveness.yaml` and `grpc-liveness.yaml` in `content/en/examples/pods/probe/` for the pattern.

Author: fixed , thanks

    ports:
    - containerPort: 8443
    livenessProbe:
      grpc:
        port: 8443
        mode: TLS
      initialDelaySeconds: 5
      periodSeconds: 10
```

> When `mode` is set to `TLS`, the kubelet connects over TLS with

Reviewer: ```suggestion
When `mode` is set to `TLS`, the `kubelet` connects over TLS with
```

Author: fixed, thanks

> `InsecureSkipVerify`, it does not verify the server certificate. This matches

Reviewer: ```suggestion
`InsecureSkipVerify` and does not verify the server certificate. This matches
```

Author: done.

> the behaviour of HTTPS HTTP probes. Certificate verification is not supported.

Reviewer: "behaviour" → "behavior" (US English). Also, "HTTPS HTTP probes" reads oddly — "HTTPS probes" is sufficient.

```suggestion
the behavior of HTTPS probes. Certificate verification is not supported.
```

Author: fixed.

> If the feature gate is disabled, the API server removes the `mode`

Reviewer: nit: Earlier in this section we use `kube-apiserver` (line 250). For consistency, use the component name here too.

```suggestion
If the feature gate is disabled, the `kube-apiserver` removes the `mode`
```

Author: fixed.
